### ISSUE TITLE
Early stopping support for GBDTs (e.g., xgboost).

### ISSUE THREAD

**Issue by kmedved** [petersontylerd/mlmachine#7]
If I want to use `mlmachine` to do a hyperparameter search of a GBDT like `xgboost` using a validation set and early stopping, is that something `mlmachine` can support?

This would require passing a different validation set into each `.fit()` call, so I'm not sure how to do this in this framework, and don't see this covered in any of the examples, but may be missing something.

Thanks.

**Comment by petersontylerd** [comment]
Thank you very much for the question.

Early stopping would be a great addition to the hyperparameter search workflow. You are correct that is not currently supported. I'd welcome a pull request if you want to contribute. Otherwise I'll investigate in the near future.

Regarding the validation set options, I think I understand what you're asking. You're thinking that it would be beneficial to be able to perform cross validation while being able to pass in a completely separate validation dataset. Do I have that right?

**Comment by petersontylerd** [comment]
The `exec_bayes_optim_search` function has been expanded to accommodate validation data being passed in along side a training dataset.  Also, the result summary now has a column to track the validation score.

**Comment by kmedved** [comment]
I apologize for not responding sooner - I missed your question 9 months ago @petersontylerd. I'll investigate the new functionality. Thanks!
